src/Surfnet/StepupMiddleware/CommandHandlingBundle/Tests/Configuration/Processor/InstitutionConfigurationProcessorTest.php

```diff
@@ -0,0 +1,349 @@
+        $institutionConfigurationProcessor->applyWhitelistCreatedEvent($whitelistCreatedEvent);
+    }
+
+    /**
+     * @test
+     * @group processor
+     * @group institution-configuration
+     */
+    public function no_create_institution_configuration_command_is_processed_for_an_already_configured_institution_when_a_whitelist_was_created()
+    {
+        $alreadyPresentInstitution = 'Already present';
+        $newInstitution            = 'New';
+
+        $whitelistCreatedEvent = new WhitelistCreatedEvent(
+            new InstitutionCollection(
+                [
+                    new Institution($alreadyPresentInstitution),
+                    new Institution($newInstitution),
+                ]
+            )
+        );
+
+        $repositoryMock = Mockery::mock(ConfiguredInstitutionRepository::class);
+        $repositoryMock->shouldReceive('hasConfigurationFor')
+            ->once()
+            ->andReturn(true);
+        $repositoryMock->shouldReceive('hasConfigurationFor')
+            ->once()
+            ->andReturn(false);
+
+        $expectedInstitution = new ConfigurationInstitution($newInstitution);
+
+        $pipelineMock = Mockery::mock(Pipeline::class);
+        $pipelineMock->shouldReceive('process')
+            ->once()
+            ->andReturnUsing(function (CreateInstitutionConfigurationCommand $command) use ($expectedInstitution) {
+                $this->assertEquals($expectedInstitution, $command->institution);
+            });
+
+        $institutionConfigurationProcessor = new InstitutionConfigurationProcessor($repositoryMock, $pipelineMock);
+        $institutionConfigurationProcessor->applyWhitelistCreatedEvent($whitelistCreatedEvent);
+    }
+
+    /**
+     * @test
+     * @group processor
+     * @group institution-configuration
+     */
+    public function create_institution_configuration_commands_are_created_when_a_whitelist_was_replaced_containing_non_configured_institutions()
+    {
+        $firstInstitutionName  = 'First institution';
+        $secondInstitutionName = 'Second institution';
+
+        $whitelistReplacedEvent = new WhitelistReplacedEvent(
+            new InstitutionCollection(
+                [
+                    new Institution($firstInstitutionName),
+                    new Institution($secondInstitutionName),
+                ]
+            )
+        );
+
+        $repositoryMock = Mockery::mock(ConfiguredInstitutionRepository::class);
+        $repositoryMock->shouldReceive('hasConfigurationFor')
+            ->andReturn(false);
+
+        $firstExpectedInstitution  = new ConfigurationInstitution($firstInstitutionName);
+        $secondExpectedInstitution = new ConfigurationInstitution($secondInstitutionName);
+
+        $pipelineMock = Mockery::mock(Pipeline::class);
+        $pipelineMock->shouldReceive('process')
+            ->once()
+            ->andReturnUsing(function (CreateInstitutionConfigurationCommand $command) use ($firstExpectedInstitution) {
```

Comment:
this is not what this method is intended for (see [the docs][mock]). Might be better to create a custom matcher. For reference on how to do something like that see [this matcher][1].

[mock]: http://docs.mockery.io/en/latest/reference/expectations.html
[1]: https://github.com/OpenConext/OpenConext-engineblock/blob/5.x-dev/tests/OpenConext/Mockery/Matcher/ValueObjectEqualsMatcher.php